tommy-server/src/metadata-cache/metadataCache.ts

```diff
@@ -0,0 +1,59 @@
+import * as redis from 'redis';
+import { promisify } from 'util';
+import axios from 'axios';
+import { config } from '../config'
+import express from 'express';
+import { AccessTokenProvider } from '../access-token/access-token-service';
+import { logger } from '../utils/logger-client';
+
+export default class MetadataCache {
+
+    private static client: redis.RedisClient = redis.createClient(config.redis.host);
+    private static getRedis = promisify(MetadataCache.client.get).bind(MetadataCache.client);
+    private static setRedis = promisify(MetadataCache.client.setex).bind(MetadataCache.client);
+
+    public static async metadataHttpCacheMiddleware(req: express.Request, res: express.Response, next: express.NextFunction) {
+
+        if (!config.lehava_api.requestTypesToCache.includes(MetadataCache.lehavaReqTypeParse(req.originalUrl))) {
+            next();
+            return;
+        }
+
+        const redisCachedResponse = JSON.parse(await MetadataCache.getRedis(req.originalUrl));
```

Comment:
if an error happens -> JSON.parse would fail. 
is this handled in outside scope? 

redis fail should be graceful.